labman/db/process.py

```diff
@@ -3237,32 +3280,32 @@ def generate_prep_information(self):
                     w1.plate_id = p1.plate_id)
                 FULL JOIN qiita.study_sample USING (sample_id)
                 WHERE sequencing_process_id = %s
-                ORDER BY study_id, sample_id, row_num, col_num, i5.barcode_seq
                 """
 
         with sql_connection.TRN as TRN:
-            # to simplify the main queries, let's get all the equipment info
+            # Let's cache some data to avoid quering the DB multiple times
+            # 1/3. equipment
             TRN.add("""SELECT equipment_id, external_id, notes, description
                        FROM labman.equipment
                        LEFT JOIN labman.equipment_type
                        USING (equipment_type_id)""")
-            equipment = {}
-            for row in TRN.execute_fetchindex():
-                row = dict(row)
-                eid = row.pop('equipment_id')
-                equipment[eid] = row
-
-            # and the reagents
+            equipment = {dict(row)['equipment_id']: dict(row)
+                         for row in TRN.execute_fetchindex()}
+            # 2/3. reagents
             TRN.add("""SELECT reagent_composition_id, composition_id,
                            external_lot_id, description
                        FROM labman.reagent_composition
                        LEFT JOIN labman.reagent_composition_type
                        USING (reagent_composition_type_id)""")
-            reagent = {}
-            for row in TRN.execute_fetchindex():
-                row = dict(row)
-                rid = row.pop('reagent_composition_id')
-                reagent[rid] = row
+            reagent = {dict(row)['reagent_composition_id']: dict(row)
+                       for row in TRN.execute_fetchindex()}
+            # 3/3. marker gene primer sets
+            TRN.add("""SELECT marker_gene_primer_set_id, primer_set_id,
+                           target_gene, target_subfragment, linker_sequence,
+                           fwd_primer_sequence, rev_primer_sequence, region
+                       FROM labman.marker_gene_primer_set""")
+            marker_gene_primer_set = {dict(row)['primer_set_id']: dict(row)
```

Comment:
same q

Reply:
...to rephrase, I'm concerned about the following type of scenario:

```python
>>> {k: i for i, k in enumerate([1,1,1,2,2,2,3,3,3,4,4,4])}
{1: 2, 2: 5, 3: 8, 4: 11}
```

I don't have the schema well in mind, so it's not clear from just looking at these queries whether the above situation is or isn't possible

Reply:
Thanks